warehouse/warehouse.go

```diff
@@ -580,11 +581,18 @@ func NewWhManager(destType string) (WarehouseManager, error) {
 
 func (wh *HandleT) mainLoop() {
 	for {
+		wh.configSubscriberLock.RLock()
 		if !wh.isEnabled {
+			wh.configSubscriberLock.RUnlock()
 			time.Sleep(mainLoopSleep)
 			continue
 		}
-		for _, warehouse := range wh.warehouses {
+		wh.configSubscriberLock.RUnlock()
+
+		configSubscriberLock.RLock()
+		warehouses := wh.warehouses
```

Comment:
Can we move this into the above lock? Would that be same?

Reply:
wh.warehouses needs configSubscriberLock and wh.isEnabled needs wh.configSubscriberLock.

configSubscriberLock and wh.configSubscriberLock are different.